.github/workflows/gcp_deploy.yaml

```diff
@@ -0,0 +1,63 @@
+name: Deploy to GCS Stage
+
+on:
+  push:
+    branches:
+      - gcs-pipeline
+
+jobs:
+  test-website:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Setup Node js
+        uses: actions/setup-node@v3
+        with:
+          node-version: "18"
+      - name: Install dependencies
+        run: yarn install --immutable
+      - name: Test current build
+        run: yarn run test-ci
+        continue-on-error: true
+
+  deploy-stage:
+    runs-on: ubuntu-latest
+    needs: test-website
+    if: ${{ success() }}
+    environment: stage
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 1
+          fetch-tags: true
+
+      - name: Setup Node js
+        uses: actions/setup-node@v3
+
+      - name: Set up Google Cloud SDK
```

Comment:
@pattersonbl2 already mentioned this was on his radar, but the only change I'd make here is using workload identify instead of passing around a service account key. https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/configuring-openid-connect-in-google-cloud-platform